### PR TITLE
Adding DnsZone Partner to New-SqlInstance cmdlet

### PR DESCRIPTION
**Pull request by ziwa-msft**
## Description

Add DnsZonePartner Parameter for New-AzureRmSqlManagedInstance cmdlet to support AutoDr for Managed Instance. Properties of the ManagedInstance model have also been updated to reflect new members DnsZone and DnsZonePartner.
This is a resubmittion of PR:https://github.com/Azure/azure-powershell/pull/7458. The original branch got deleted by mistake during maintenance.

## Checklist

- [x ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x ] The title of the PR is clear and informative
- [x ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x ] If applicable, the changes made in the PR have proper test coverage
- [x ] For public API changes to cmdlets:
    - [ x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs [hunk 1]
```diff
@@ -213,6 +213,13 @@ public class NewAzureSqlManagedInstance : ManagedInstanceCmdletBase
             HelpMessage = "Generate and assign an Azure Active Directory Identity for this instance for use with key management services like Azure KeyVault.")]
         public SwitchParameter AssignIdentity { get; set; }
 
+        /// <summary>
+        /// Gets or sets the managed instance compute generation
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The Dns Zone Partner Resource ID for the Sql Azure Managed Instance.")]
+        public string DnsZonePartner { get; set; }
```
Comment (markcowl) [inline src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs]: Are there any known values for this, or is this always free-form text?

File: src/Sql/Sql.Test/ScenarioTests/ManagedInstanceCrudScenarioTests.ps1 [hunk 2]
```diff
@@ -68,11 +68,12 @@ function Test-CreateManagedInstance
 		$edition = "GeneralPurpose"
 		$computeGeneration = "Gen4"
 		$managedInstanceName = Get-ManagedInstanceName
+		$dnsZonePartner = $managedInstance1.ResourceId
```
Comment (markcowl) [inline src/Sql/Sql.Test/ScenarioTests/ManagedInstanceCrudScenarioTests.ps1]: You should verify that this value comes back, and re-record the test for the new value

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs (hunk 1 above)
Comment (ziwa-msft) [reply src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs]: There is no known value. it is an ARM resource id

Comment (markcowl) [reply src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs]: How does the user discover the value of this resourceId?  Seems like we should decorate this property with the ResourceId Completer: https://github.com/Azure/azure-powershell/blob/1bcbe7b1f7a3323ac98f7754ba03eeb6b45e79f2/src/Compute/Compute/VirtualMachine/Action/VirtualMachineActionBaseCmdlet.cs

Comment (ziwa-msft) [reply src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs]: Added the decorator.

File: src/Sql/Sql.Test/ScenarioTests/ManagedInstanceCrudScenarioTests.ps1 (hunk 2 above)
Comment (ziwa-msft) [reply src/Sql/Sql.Test/ScenarioTests/ManagedInstanceCrudScenarioTests.ps1]: for some reason the test is passing locally with the new record session with playback mode but not in the check-in automation.

**Comment by MiYanni** [comment]
@ziwa-msft You have merge conflicts. And, some failures to investigate. I'd merge in the latest `master` and resolve any failures after it runs.

**Comment by ziwa-msft** [comment]
Can we merge this one?

**Pull request merged**